src/access/acp.test.ts

```diff
@@ -3961,3 +4064,2232 @@ describe("getAgentAccessAll", () => {
+        append: false,
+        write: false,
+        controlRead: true,
+        controlWrite: false,
+      });
+    });
+
+    it("preserves conflicting Control access defined for a different actor that is defined with the same Rule as applies to the given actor", () => {
+      const resourceWithAcr = mockResourceWithAcr(
+        "https://some.pod/resource",
+        "https://some.pod/resource?ext=acr",
+        {
+          policies: {},
+          memberPolicies: {},
+          acrPolicies: {
+            "https://some.pod/resource?ext=acr#policy": {
+              deny: { read: true },
+              allOf: {
+                "https://some.pod/resource?ext=acr#rule": {
+                  [acp.agent]: [
+                    webId,
+                    "https://some-other.pod/other-profile#me",
+                  ],
+                },
+              },
+            },
+          },
+          memberAcrPolicies: {},
+        }
+      );
+
+      const updatedResourceWithAcr = internal_setActorAccess(
+        resourceWithAcr,
+        acp.agent,
+        webId,
+        {
+          controlRead: true,
+        }
+      );
+
+      expect(
+        internal_getActorAccess(
+          resourceWithAcr,
+          acp.agent,
+          "https://some-other.pod/other-profile#me"
+        )
+      ).toStrictEqual({
+        controlRead: false,
+      });
+      expect(
+        internal_getActorAccess(updatedResourceWithAcr!, acp.agent, webId)
+      ).toStrictEqual({
+        controlRead: true,
+      });
+    });
+
+    it("preserves conflicting access defined for a different actor that is defined with the same Rule as applies to the given actor", () => {
+      const resourceWithAcr = mockResourceWithAcr(
+        "https://some.pod/resource",
+        "https://some.pod/resource?ext=acr",
+        {
+          policies: {
+            "https://some.pod/resource?ext=acr#policy": {
+              deny: { read: true },
+              allow: { append: true },
+              allOf: {
+                "https://some.pod/resource?ext=acr#rule": {
+                  [acp.agent]: [
+                    webId,
+                    "https://some-other.pod/other-profile#me",
+                  ],
+                },
+              },
+            },
+          },
+          memberPolicies: {},
+          acrPolicies: {},
+          memberAcrPolicies: {},
+        }
+      );
+
+      const updatedResourceWithAcr = internal_setActorAccess(
+        resourceWithAcr,
+        acp.agent,
+        webId,
+        {
+          read: true,
+        }
+      );
+
+      expect(
+        internal_getActorAccess(
+          resourceWithAcr,
+          acp.agent,
+          "https://some-other.pod/other-profile#me"
+        )
+      ).toStrictEqual({
+        read: false,
+        append: true,
+      });
+      expect(
+        internal_getActorAccess(updatedResourceWithAcr!, acp.agent, webId)
+      ).toStrictEqual({
+        read: true,
+        append: true,
```

Comment:
Ah hmm, I see what you mean now. So that is what the `expect` at line 4609 does; this `expect` merely checks that the correct access _is_ set for the the agent `webId`, but I suppose that that is not the point of this test and is already covered by other test. I'll remove the `expect` both here and with the previous one, does that make sense?

Reply:
Or maybe I should just add "while setting the appropriate access for the given actor" to the test description, just to make sure?

Reply:
I think the change in 9f77b87 is sufficient, but keeping this test to double check that  having a contradictory `Access` set initially doesn't prevent proper access to be set may also make sense, up to you.